sdk/dotnet/Pulumi/Pulumi.csproj

```diff
@@ -9,14 +11,16 @@
     <RepositoryUrl>https://github.com/pulumi/pulumi</RepositoryUrl>
     <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
     <PackageIcon>pulumi_logo_64x64.png</PackageIcon>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
 
-    <TargetFramework>netcoreapp3.0</TargetFramework>
-    <Nullable>enable</Nullable>
+  <PropertyGroup>
+    <NoWarn>NU5105</NoWarn>
```

Comment:
Curious why this change?

Reply:
I wasn't sure why we were only setting it for `Debug|AnyCPU`. This sets it for both Release and Debug configurations like `Pulumi.FSharp`.

Reply:
Do we even do Release builds?

Reply:
No :)

Reply:
> No :)

By design? I'm happy to revert this part of the change if it's unnecessary.

Reply:
I think we just never discussed this. I suspect we don't want to use semver 2.0 for release versions but I may be wrong.